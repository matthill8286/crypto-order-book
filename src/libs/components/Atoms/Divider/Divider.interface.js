
//# sourceMappingURL=Divider.interface.js.map