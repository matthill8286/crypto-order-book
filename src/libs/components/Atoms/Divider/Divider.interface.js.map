{"version":3,"sources":[],"names":[],"mappings":"","sourcesContent":[],"file":"Divider.interface.js"}