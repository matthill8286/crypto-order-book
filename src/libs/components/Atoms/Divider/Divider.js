function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import * as React from 'react';
import { css, styled } from "../../../styles/styled";
import { getSizeInPx } from "../../../styles";
var DefaultSize = 'xs';
var StyledHR = styled.hr.withConfig({
  displayName: "Divider__StyledHR",
  componentId: "sc-40hr1c-0"
})(function (_ref) {
  var theme = _ref.theme,
      color = _ref.color,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? DefaultSize : _ref$height;
  return css(["height:", ";width:100%;background:", ";border-width:0;margin:0;"], getSizeInPx(height, theme.spacing.base), theme.color[color]);
});
export var Divider = function Divider(_ref2) {
  var color = _ref2.color,
      height = _ref2.height,
      otherProps = _objectWithoutProperties(_ref2, ["color", "height"]);

  return React.createElement(StyledHR, _extends({
    height: height,
    color: color
  }, otherProps));
};
//# sourceMappingURL=Divider.js.map