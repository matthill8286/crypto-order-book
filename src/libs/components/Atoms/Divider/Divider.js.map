{"version":3,"sources":["../../../../src/components/Atoms/Divider/Divider.tsx"],"names":["React","css","styled","getSizeInPx","DefaultSize","StyledHR","hr","theme","color","height","spacing","base","Divider","otherProps"],"mappings":";;;;;;AAAA,OAAO,KAAKA,KAAZ,MAAuB,OAAvB;AACA,SAASC,GAAT,EAAcC,MAAd;AAEA,SAASC,WAAT;AAOA,IAAMC,WAAW,GAAG,IAApB;AAEA,IAAMC,QAAQ,GAAGH,MAAM,CAACI,EAAV;AAAA;AAAA;AAAA,GACZ;AAAA,MAAGC,KAAH,QAAGA,KAAH;AAAA,MAAUC,KAAV,QAAUA,KAAV;AAAA,yBAAiBC,MAAjB;AAAA,MAAiBA,MAAjB,4BAA0BL,WAA1B;AAAA,SAA4CH,GAA5C,sEACYE,WAAW,CAACM,MAAD,EAASF,KAAK,CAACG,OAAN,CAAcC,IAAvB,CADvB,EAGgBJ,KAAK,CAACC,KAAN,CAAYA,KAAZ,CAHhB;AAAA,CADY,CAAd;AAUA,OAAO,IAAMI,OAA+B,GAAG,SAAlCA,OAAkC,QAAsC;AAAA,MAAnCJ,KAAmC,SAAnCA,KAAmC;AAAA,MAA5BC,MAA4B,SAA5BA,MAA4B;AAAA,MAAjBI,UAAiB;;AACnF,SAAO,oBAAC,QAAD;AAAU,IAAA,MAAM,EAAEJ,MAAlB;AAA0B,IAAA,KAAK,EAAED;AAAjC,KAA4CK,UAA5C,EAAP;AACD,CAFM","sourcesContent":["import * as React from 'react'\nimport { css, styled } from '@/styles/styled'\nimport { Size, ThemeColors } from '@/types/theme'\nimport { getSizeInPx } from '@/styles'\n\nexport interface DividerProps {\n  color: ThemeColors\n  height?: number | keyof Omit<Size, 'xxxs'>\n}\n\nconst DefaultSize = 'xs'\n\nconst StyledHR = styled.hr<DividerProps>(\n  ({ theme, color, height = DefaultSize }) => css`\n    height: ${getSizeInPx(height, theme.spacing.base)};\n    width: 100%;\n    background: ${theme.color[color]};\n    border-width: 0;\n    margin: 0;\n  `\n)\n\nexport const Divider: React.FC<DividerProps> = ({ color, height, ...otherProps }) => {\n  return <StyledHR height={height} color={color} {...otherProps} />\n}\n"],"file":"Divider.js"}