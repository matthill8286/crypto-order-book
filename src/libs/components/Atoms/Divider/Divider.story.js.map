{"version":3,"sources":["../../../../src/components/Atoms/Divider/Divider.story.tsx"],"names":["select","storiesOf","React","saiyanTheme","Divider","options","Object","keys","color","height","knobs","module","add"],"mappings":"AAAA,SAAiBA,MAAjB,QAA+B,wBAA/B;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,WAAT;AAEA,SAASC,OAAT;AAEA,IAAMC,OAAO,GAAGC,MAAM,CAACC,IAAP,CAAYJ,WAAW,CAACK,KAAxB,CAAhB;AACA,IAAMC,MAAM,GAAG,CAAC,CAAD,EAAI,EAAJ,EAAQ,IAAR,EAAc,EAAd,EAAkB,EAAlB,CAAf;;AAEA,IAAMC,KAAK,GAAG,SAARA,KAAQ,GAAM;AAClB,SAAO;AACLF,IAAAA,KAAK,EAAER,MAAM,CAAC,OAAD,EAAUK,OAAV,EAAmB,SAAnB,CADR;AAELI,IAAAA,MAAM,EAAET,MAAM,CAAC,QAAD,EAAWS,MAAX,EAAmBA,MAAM,CAAC,CAAD,CAAzB;AAFT,GAAP;AAID,CALD;;AAOAR,SAAS,CAAC,6BAAD,EAAgCU,MAAhC,CAAT,CAAiDC,GAAjD,CAAqD,SAArD,EAAgE,YAAM;AACpE,SAAO,oBAAC,OAAD,EAAaF,KAAK,EAAlB,CAAP;AACD,CAFD","sourcesContent":["import { number, select } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { saiyanTheme } from '@/styles/sc-vars-saiyan'\nimport { ThemeColors } from '@/types/theme'\nimport { Divider } from './Divider'\n\nconst options = Object.keys(saiyanTheme.color) as ThemeColors[]\nconst height = [5, 10, 12.5, 13, 15]\n\nconst knobs = () => {\n  return {\n    color: select('Color', options, 'primary'),\n    height: select('Height', height, height[5]),\n  }\n}\n\nstoriesOf('Design System/Atoms/Divider', module).add('Default', () => {\n  return <Divider {...knobs()} />\n})\n"],"file":"Divider.story.js"}