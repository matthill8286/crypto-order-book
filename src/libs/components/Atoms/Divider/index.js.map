{"version":3,"sources":["../../../../src/components/Atoms/Divider/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './Divider'\n"],"file":"index.js"}