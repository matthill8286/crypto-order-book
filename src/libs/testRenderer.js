import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map