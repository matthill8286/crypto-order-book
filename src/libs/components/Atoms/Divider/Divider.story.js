import { select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { saiyanTheme } from "../../../styles/sc-vars-saiyan";
import { Divider } from "./Divider";
var options = Object.keys(saiyanTheme.color);
var height = [5, 10, 12.5, 13, 15];

var knobs = function knobs() {
  return {
    color: select('Color', options, 'primary'),
    height: select('Height', height, height[5])
  };
};

storiesOf('Design System/Atoms/Divider', module).add('Default', function () {
  return React.createElement(Divider, knobs());
});
//# sourceMappingURL=Divider.story.js.map