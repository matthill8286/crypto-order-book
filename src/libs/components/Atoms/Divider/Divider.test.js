import React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { Divider } from "./Divider";
describe('Divider', function () {
  it('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(Divider, {
      color: "primary"
    }));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=Divider.test.js.map