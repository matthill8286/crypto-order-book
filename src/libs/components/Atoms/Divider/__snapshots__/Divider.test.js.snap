// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly 1`] = `
<hr
  className="Divider__StyledHR-sc-40hr1c-0 guSWuT"
  color="primary"
/>
`;
